R/length-params.R

```diff
@@ -0,0 +1,158 @@
+
+  logger::log_info("Getting taxa rank...")
+  ranks <- taxize::tax_rank(spp$name_scientific,db = "gbif",rows=1)
+
+
+  ranks_tab <-
+    ranks %>%
+    purrr::map(~stringr::str_to_title(.x)) %>%
+    purrr::map(~tidyr::as_tibble(.x)) %>%
+    purrr::map_dfr(~dplyr::mutate_all(.x,as.character)) %>%
+    dplyr::rename(taxa_rank=value) %>%
+    dplyr::bind_cols(spp,.) %>%
+    dplyr::mutate(name=name_scientific) %>%
+    dplyr::mutate(name=dplyr::case_when(name=="Thunnini"~"Tuna",
+                                        name=="Rajiformes"~"Manta",
+                                        TRUE~.data$name),
+                  taxa_rank=dplyr::case_when(name=="Tuna"~"comm_name",
+                                             name=="Manta"~"comm_name",
+                                             TRUE~.data$taxa_rank)) %>%
+    dplyr::filter(!catch_name_en %in% c("No catch","Sea cucumber",
+                                        "Other","Unknown","Lobster"))
+
+  ranks_tab
+}
+
+
+#' Download species body-length parameters
+#'
+#' This function takes species, group of species or common names as arguments and
+#' return a data frame containing body-length parameters associated to the
+#' FishBase database.
+#'
+#' @param taxa The name of the species/groups from which retrieve lengths parameters.
+#' @param rank The taxonomical ranking, can be `Species`, `Genus`, `Family`, or
+#' `comm_name.` The latter is an abbreviation for the  common name of a species/group.
+#' @param country_code The code of the country associated to FishBase database.
+#' @param get_one Logical, indicating wether the user wants to get length data
+#' from the most recorded species.
+#'
+#' @return A data frame containing species length parameters
+#' @export
+#'
+#' @examples
+#' get_fish_length(taxa="Sardine",rank="comm_name",
+#' country_code=626,get_one=FALSE)
+#'
+get_fish_length <- function(taxa,
+                            rank=NULL,
+                            country_code=NULL,
+                            get_one=NULL){
+
+  take_one <- function(specs_length){
+    specs_length %>%
+      dplyr::arrange(desc(Number)) %>%
+      dplyr::slice(1)
+  }
+
+  if (rank=="Species"){
+    specs_length <- rfishbase::length_length(taxa)
+    return(specs_length)
+  }
+
+  if (rank=="Genus"){
+    taxa <- stringr::word(taxa, 1)
+    specs <-
+      rfishbase::country(rfishbase::species_list(Genus=taxa)) %>%
+      dplyr::filter(C_Code==country_code) %>%
+      magrittr::extract2("Species")
+  }
+
+  if (rank=="Family"){
+    specs <-
+      rfishbase::country(rfishbase::species_list(Family=taxa)) %>%
+      dplyr::filter(C_Code==country_code) %>%
+      magrittr::extract2("Species")
+  }
+
+  if (rank=="comm_name"){
+    sp_list <-
+      rfishbase::common_to_sci(taxa) %>% magrittr::extract2("Species")
+    specs <-
+      rfishbase::country(sp_list) %>%dplyr::filter(C_Code==country_code)%>%
+      magrittr::extract2("Species")
+  }
```

Comment:
I haven't tested it, but I think you could avoid most of the if statements (except the common name one) by executing a function call

```suggestion
  if (rank == "comm_name") {
    sp_list <- rfishbase::common_to_sci(taxa) %>%
      magrittr::extract2("Species")
  } else {
    sp_list <- list(taxa) %>%
      rlang::set_names(rank) %>%
      do.call(rfishbase::species_list, .)
  }

  specs <- sp_list %>%
    rfishbase::country() %>%
    dplyr::filter(C_Code %in% country_code) %>%
    magrittr::extract2("Species")
```

Reply:
Yeah @efcaguab, that's a great tip!